evap/evaluation/models.py

```diff
@@ -711,9 +723,12 @@ def editor_approve(self):
     def manager_approve(self):
         pass
 
-    @transition(field=state, source=[State.PREPARED, State.EDITOR_APPROVED, State.APPROVED], target=State.NEW)
-    def revert_to_new(self):
-        pass
+    @transition(field=state, target=State.NEW, conditions=[lambda self: self.can_reset_to_new])
+    def reset_to_new(self, *, delete_previous_answers: bool):
+        if delete_previous_answers:
+            for answer_class in Answer.__subclasses__():
+                answer_class._default_manager.filter(contribution__evaluation_id=self.id).delete()
```

Comment:
Does the following also work?
```suggestion
                answer_class._default_manager.filter(contribution__evaluation=self).delete()
```